src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsContextualizationTest.java

```diff
@@ -0,0 +1,99 @@
+                        true,
+                        false))
+                .willReturn(aResponse()
+                        .withHeader("Content-Type", "application/json; charset=utf-8")
+                        .withBody("{\"token\":\"super-secret-token\",\"expires_at\":\"" + createTokenExpiration()
+                                + "\"}")));
+
+        // Add wiremock responses for Repository
+        githubApi.stubFor(get(urlEqualTo("/repos/cloudbeers/multibranch-demo"))
+                .willReturn(aResponse()
+                        .withHeader("Content-Type", "application/json; charset=utf-8")
+                        .withBodyFile("../contextualization/body-repos-cloudbeers-multibranch-demo.json")));
+        githubApi.stubFor(get(urlEqualTo("/repos/cloudbeers/multibranch-demo/branches"))
+                .willReturn(aResponse()
+                        .withHeader("Content-Type", "application/json; charset=utf-8")
+                        .withBodyFile("../contextualization/body-repos-cloudbeers-multibranch-demo-branches.json")));
+        githubApi.stubFor(get(urlEqualTo("/repos/cloudbeers/multibranch-demo/contents/?ref=refs%2Fheads%2Fmaster"))
+                .willReturn(aResponse()
+                        .withHeader("Content-Type", "application/json; charset=utf-8")
+                        .withBodyFile("../contextualization/body-repos-cloudbeers-multibranch-demo-contents.json")));
+        githubApi.stubFor(get(urlEqualTo("/repos/cloudbeers/multibranch-demo/pulls?state=open"))
+                .willReturn(aResponse()
+                        .withHeader("Content-Type", "application/json; charset=utf-8")
+                        .withBodyFile("../contextualization/body-repos-cloudbeers-multibranch-demo-pulls.json")));
+    }
+
+    @Test
+    public void ownerMustBeInferedFromRepository() throws Exception {
+        final var store = CredentialsProvider.lookupStores(r.jenkins).iterator().next();
+
+        final var credentials = GitHubApp.createCredentials("myAppCredentialsWithoutOwner");
+        store.addCredentials(Domain.global(), credentials);
+        credentials.setApiUri(githubApi.baseUrl());
+
+        final var scmSource = new GitHubSCMSource("cloudbeers", "multibranch-demo", null, false);
+        scmSource.setTraits(Arrays.asList(new BranchDiscoveryTrait(true, true)));
+        scmSource.setCredentialsId(credentials.getId());
+        scmSource.setApiUri(githubApi.baseUrl());
+
+        final var multiBranchProject = r.jenkins.createProject(WorkflowMultiBranchProject.class, "multibranch-demo");
+        multiBranchProject.getSourcesList().add(new BranchSource(scmSource));
+        multiBranchProject.scheduleBuild2(0, new Action[0]).getFuture().get();
+
+        final var branchProject = multiBranchProject.getItem("master");
+        assertThat(branchProject, notNullValue());
```

Comment:
And without the fix, is this the line that fails?

Reply:
> And without the fix, is this the line that fails?

Yes. I confirm that the test is failing without https://github.com/jenkinsci/github-branch-source-plugin/pull/803